decidim-dev/lib/decidim/dev/test/rspec_support/capybara.rb

```diff
@@ -81,7 +81,7 @@ def switch_to_secure_context_host
   )
 end
 
-Capybara.server = :puma, { Silent: true, Threads: "1:1" }
+Capybara.server = :puma, { Silent: true, queue_requests: false }
```

Comment:
I had to disable the Puma `queue_requests` option because otherwise e.g. this spec was failing pretty constantly (i.e. failed more often than passed):
https://github.com/decidim/decidim/blob/95dc6673425d5b7f2d30f3b0c24dcda8aeb5d88a/decidim-proposals/spec/system/filter_proposals_spec.rb#L515

The reason for that was that the login call was not working properly in the `before` hook:
https://github.com/decidim/decidim/blob/95dc6673425d5b7f2d30f3b0c24dcda8aeb5d88a/decidim-proposals/spec/system/filter_proposals_spec.rb#L509-L510

When the test user requested the page, they were not logged in for one reason or another.

The `login_as` method is provided by Warden:
https://github.com/wardencommunity/warden/blob/8b44aa4dd8ff4bfcc0e7257dea095789ffb58690/lib/warden/test/helpers.rb#L18-L23

What it does is that it adds a hook for the "next reqeust" to be run prior to serving that request. Apparently there was some queued request that already consumed the login state when the `queue_requests` option was enabled (default).

After disabling it, the described spec seems to be passing constantly.

Here's some more information about how Puma handles the requests in multi-threaded mode with and without this option:
https://github.com/puma/puma/blob/master/docs/architecture.md